### PR TITLE
TM Add Unit Tests Part1

### PR DESCRIPTION
**Pull request by AYUSH325**
## Description

*What was completed, changed, or updated?* 

Added Mocking instances and mock methods for TM manager.

<br/>

*Why was this done (if applicable)?*

To set the basis for writing unit tests for TM on completion of the TM classes.

<br/>

## Testing

*What **manual** tests were used to validate the code?*

N/A

<br/>

*What **unit** tests were used to validate the code?*

N/A

<br/>

## Documentation

*Milestone number and name:*

*Link to Asana task:* https://app.asana.com/0/1203458353737758/1207155215118029

*Link to Confluence documentation:*

<br/>

## Reminders

- [x] Add reviewers to the PR

- [x] Mention the PR in the appropriate discord channel

### CODE REVIEW

**Review by Yaremadzulynsky — COMMENTED** [review]
everything LGTM, but other the mocks seem to be in ${workspace}/Testing/Mocks/...

Can you check with one of the leads to see if they want Mocks in the TM directory or in the Testing directory? After that I'm good to approve.

**Comment by AYUSH325** [comment]
> everything LGTM, but other the mocks seem to be in ${workspace}/Testing/Mocks/...
> 
> Can you check with one of the leads to see if they want Mocks in the TM directory or in the Testing directory? After that I'm good to approve.

@HardyYu @DerekTang04

File: TelemetryManager/Tests/Mocks/Inc/MockMavlinkTranslator.hpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#include "gmock/gmock.h"
+#include "MavlinkTranslator.hpp"
+#include "TMCircularBuffer.hpp"
+#include "Official_Mavlink_2_Library/common/mavlink.h"
+
+class MockMavlinkTranslator : MavlinkTranslator
+{
+    MOCK_METHOD(void, bytesToMavlinkMsg, (TMCircularBuffer buffer));
+    MOCK_METHOD(void, addMavlinkMsgToByteQueue, (mavlink_message_t msg, TMCircularBuffer buffer));
+};
```
Comment (HardyYu) [inline TelemetryManager/Tests/Mocks/Inc/MockMavlinkTranslator.hpp:10]: Adding one empty line to the bottom of a file can help with passing the GCC compiling. https://www.unix.com/programming/137233-why-required-leave-empty-line-end-c-program.html

**Review by HardyYu — APPROVED** [review]
LGTM! I have some minor comments there that doesn't really matter. I think the directory where you write this mock makes sense.

**Review by Yaremadzulynsky — APPROVED** [review]

**Pull request merged**